sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2120,6 +2120,8 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     }
   }
 
+  private final val alphabet = "[a-zA-Z]".r.pattern
```

Comment:
instead of checking the invalid ones, how about we make sure the value only has digits and dots?

Reply:
OK updated.